e2e/snapshot_ops_test.go

```diff
@@ -0,0 +1,234 @@
+	createReq := api.VolCreateReq{
+		Name: volname,
+		Subvols: []api.SubvolReq{
+			{
+				ReplicaCount: 2,
+				Type:         "replicate",
+				Bricks: []api.BrickReq{
+					{PeerID: gds[0].PeerID(), Path: brickPaths[0]},
+					{PeerID: gds[1].PeerID(), Path: brickPaths[1]},
+				},
+			},
+			{
+				Type:         "replicate",
+				ReplicaCount: 2,
+				Bricks: []api.BrickReq{
+					{PeerID: gds[0].PeerID(), Path: brickPaths[2]},
+					{PeerID: gds[1].PeerID(), Path: brickPaths[3]},
+				},
+			},
+		},
+		Force: true,
+	}
+	_, err := client.VolumeCreate(createReq)
+	return err
+}
+
+func volumeDelete(client *restclient.Client) error {
+	return client.VolumeDelete(volname)
+}
+
+func testSnapshotCreate(t *testing.T) {
+	r := require.New(t)
+	snapshotCreateReq := api.SnapCreateReq{
+		VolName:  volname,
+		SnapName: snapname,
+	}
+	_, err := client.SnapshotCreate(snapshotCreateReq)
+	r.Nil(err, "snapshot create failed")
+
+	snapshotCreateReq = api.SnapCreateReq{
+		VolName:     volname,
+		SnapName:    snapname,
+		TimeStamp:   true,
+		Description: "Snapshot for testing, timestamp and force flags are enabled",
+		Force:       true,
+	}
+	_, err = client.SnapshotCreate(snapshotCreateReq)
+	r.Nil(err, "snapshot create failed")
+
+}
+
+func volumeStart(client *restclient.Client) error {
+
+	return client.VolumeStart(volname, true)
+}
+
+func volumeStop(client *restclient.Client) error {
+
+	return client.VolumeStop(volname)
+}
+
+func testSnapshotList(t *testing.T) {
+	r := require.New(t)
+
+	snaps, err := client.SnapshotList("")
+	r.Nil(err)
+	r.Len(snaps[0].SnapName, 2)
+	snaps, err = client.SnapshotList(volname)
+	r.Nil(err)
+	r.Len(snaps[0].SnapName, 2)
+
+}
+
+func testSnapshotInfo(t *testing.T) {
+	r := require.New(t)
+
+	//gets the info of the snapname without timestamp
+	_, err := client.SnapshotInfo(snapname)
+	r.Nil(err)
+}
+
+func testSnapshotActivate(t *testing.T) {
+	var snapshotActivateReq api.SnapActivateReq
+	r := require.New(t)
+
+	vols, err := client.SnapshotList(volname)
+	r.Nil(err)
+
+	for _, snaps := range vols {
+		for _, snapName := range snaps.SnapName {
+			err = client.SnapshotActivate(snapshotActivateReq, snapName)
+			r.Nil(err)
+
+			snapshotActivateReq.Force = true
+		}
+	}
+
+}
+
+func testSnapshotDelete(t *testing.T) {
+	r := require.New(t)
+
+	vols, err := client.SnapshotList(volname)
+	r.Nil(err)
+	r.Len(vols[0].SnapName, 2)
+
+	for _, snaps := range vols {
+		for _, snapName := range snaps.SnapName {
+			err = client.SnapshotDelete(snapName)
+			r.Nil(err)
+		}
+	}
+
+	vols, err = client.SnapshotList(volname)
+	r.Nil(err)
+	r.Len(vols[0].SnapName, 0)
+}
+
+func testSnapshotDeactivate(t *testing.T) {
+	var snapshotActivateReq api.SnapActivateReq
+	r := require.New(t)
+	vols, err := client.SnapshotList(volname)
+	r.Nil(err)
+	for _, snaps := range vols {
+		for _, snapName := range snaps.SnapName {
+			err = client.SnapshotDeactivate(snapName)
+			r.Nil(err)
+
+			snapshotActivateReq.Force = true
```

Comment:
snapshotActivateReq.Force = true is this required here, or can we move it to begin of the test case